app/Repositories/Organization/OrganizationRepository.php

```diff
@@ -0,0 +1,413 @@
+
+            $is_updated = $organization->update(Arr::except($data, ['admins', 'users']));
+            // update the organization data
+            if ($is_updated) {
+                $organization->users()->sync($userRoles);
+                DB::commit();
+            }
+
+            return $is_updated;
+        } catch (\Exception $e) {
+            Log::error($e->getMessage());
+        }
+    }
+
+    /**
+     * Delete suborganization
+     *
+     * @param $id
+     * @return Model
+     */
+    public function deleteSuborganization($id)
+    {
+        $organization = $this->find($id);
+
+        try {
+            foreach ($organization->children as $suborganization) {
+                $this->deleteSuborganization($suborganization->id);
+            }
+
+            $is_deleted = $this->delete($id);
+            return $is_deleted;
+        } catch (\Exception $e) {
+            Log::error($e->getMessage());
+        }
+    }
+
+    /**
+     * Get the member options to add in specific suborganization
+     *
+     * @param $data
+     * @param Organization $organization
+     * @return mixed
+     */
+    public function getMemberOptions($data, $organization)
+    {
+        $organizationUserIds = $organization->users()->wherePivot('organization_role_type_id', '<>', 1)->get()->modelKeys();
+        $organizationAdminUserIds = $organization->users()->wherePivot('organization_role_type_id', 1)->get()->modelKeys();
+        $parentOrganizationUserIds = $this->getParentOrganizationUserIds([], $organization, $organizationAdminUserIds);
+
+        if ($data['page'] === 'create') {
+            $userInIds = array_merge($organizationUserIds, $parentOrganizationUserIds);
+        } else if ($data['page'] === 'update') {
+            $userInIds = array_diff($parentOrganizationUserIds, $organizationUserIds);
+        }
+
+        $this->query = $this->userRepository->model->when($data['query'] ?? null, function ($query) use ($data) {
+            $query->search(['email'], $data['query']);
+            return $query;
+        });
+
+        return $this->query->whereIn('id', $userInIds)->orderBy('first_name', 'asc')->paginate();
+    }
+
+    /**
+     * Get the parent organizations user ids
+     *
+     * @param $userIds
+     * @param $organization
+     * @param $organizationAdminUserIds
+     * @return array
+     */
+    public function getParentOrganizationUserIds($userIds, $organization, $organizationAdminUserIds)
+    {
+        if ($parentOrganization = $organization->parent) {
+            $ids = $parentOrganization->users()->wherePivot('organization_role_type_id', '<>', 1)->get()->modelKeys();
+            $adminIds = $parentOrganization->users()->wherePivot('organization_role_type_id', 1)->get()->modelKeys();
+            $ids = array_diff($ids, $organizationAdminUserIds);
+            $organizationAdminUserIds = array_merge($organizationAdminUserIds, $adminIds);
+            $userIds = array_merge($userIds, $ids);
+
```

Comment:
Use double equal (==) - logical error.

Reply:
Nope, its not comparison but assignment and check for null.